cmd/cacher/main.go

```diff
@@ -65,19 +69,30 @@ func cache() error {
 		GID:          gid,
 	}
 
-	factory, err := image.NewFactory(image.WithOutWriter(os.Stdout))
-	if err != nil {
-		return err
-	}
+	var cacherCache lifecycle.Cache
+	if cacheImageTag != "" {
+		factory, err := image.NewFactory(image.WithOutWriter(os.Stdout))
+		if err != nil {
+			return err
+		}
 
-	origCacheImage, err := factory.NewLocal(cacheImageTag)
-	if err != nil {
-		return err
+		origCacheImage, err := factory.NewLocal(cacheImageTag)
+		if err != nil {
+			return err
+		}
+
+		cacherCache = cache.NewImageCache(cacher.Out, factory, origCacheImage)
```

Comment:
cacherCache feels a little awkward, maybe cacheStore?

Reply:
Agreed. Will think of something different.

Reply:
Done.